### ISSUE TITLE
Gradients returning None when using dense image warp

### ISSUE THREAD

**Issue by LucasKirsten** [tensorflow/addons#2266]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow version and how it was installed (source or binary): 2.3.0-rc1, installed via pip
- TensorFlow-Addons version and how it was installed (source or binary): 0.11.2, installed via pip
- Python version: 3.6.9
- Is GPU used? (yes/no): yes

**Describe the bug**

The gradients returned for all variables are None if using the dense_image_warp function inside a Keras model.

**Code to reproduce the issue**

```
import tensorflow as tf
import tensorflow_addons as tfa

@tf.function
def image_warp(images, flows):
    
    h,w = images.shape[1:3]
    denorm = w/10.
    
    flows = tf.image.resize(flows, (h,w))
    flows = tf.math.floor(flows * denorm)
    dense_img_warp = tfa.image.dense_image_warp(images, flows)

    return dense_img_warp

model = tf.keras.Sequential([
    tf.keras.layers.Conv2D(2, (3,3), strides=(2,2), input_shape=(224,224,3))
])

x, y = tf.random.normal((1,224,224,3)), tf.ones((1,224,224,3))

with tf.GradientTape() as tape:
    pf = model(x)
    p = image_warp(x, pf)
    loss = tf.keras.losses.mae(y, p)
    
grads = tape.gradient(loss, model.trainable_variables)

```

**Other info / logs**

Tensorflow env info: [tf_env.txt](https://github.com/tensorflow/addons/files/5642751/tf_env.txt)

**Comment by LucasKirsten** [comment]
Discovered that the problem was actually on the tf.math.floor operation.
